src/containerapp/azext_containerapp/containerapp_sessionpool_decorator.py

```diff
@@ -0,0 +1,403 @@
+
+    def get_environment_client(self):
+        return ManagedEnvironmentClient
+
+
+class SessionPoolCreateDecorator(SessionPoolPreviewDecorator):
+    def validate_arguments(self):
+        validate_container_app_name(self.get_argument_name(), AppType.SessionPool.name)
+        container_type = self.get_argument_container_type()
+        environment_name = self.get_argument_managed_env()
+
+        location = self.get_argument_location()
+        _ensure_location_allowed(self.cmd, location, CONTAINER_APPS_RP, "sessionPools")
+
+        if container_type == ContainerType.CustomContainer.name:
+            if environment_name is None:
+                raise RequiredArgumentMissingError(
+                    f'Must provide environment name when container type is {ContainerType.CustomContainer.name}')
+        else:
+            if environment_name is not None:
+                raise ValidationError(f"Do not pass environment name when using container type {container_type}")
+
+    def construct_payload(self):
+        self.session_pool_def["location"] = self.get_argument_location()
+
+        # We only support 'Dynamic' type in CLI
+        self.session_pool_def["properties"]["poolManagementType"] = "Dynamic"
+        self.session_pool_def["properties"]["environmentId"] = self.get_argument_managed_env()
+        if self.get_argument_container_type() is None:
+            self.set_argument_container_type(ContainerType.PythonLTS.name)
+        self.session_pool_def["properties"]["containerType"] = self.get_argument_container_type()
+
+        dynamic_pool_def = self.set_up_dynamic_configuration()
+        session_network_def = self.set_up_network_configuration()
+        session_scale_def = self.set_up_scale_configuration()
+        secrets_def = self.set_up_secrets()
+
+        # CustomerContainerTemplate
+        customer_container_template = None
+        if self.get_argument_container_type() == ContainerType.CustomContainer.name:
+            customer_container_template = {}
+            self.validate_environment()
+
+            container_def = self.set_up_container()
+            ingress_def = self.set_up_ingress()
+            registry_def = self.set_up_registry_auth_configuration(secrets_def)
+
+            customer_container_template["containers"] = [container_def]
+            customer_container_template["ingress"] = ingress_def
+            customer_container_template["registryCredentials"] = registry_def
+
+        safe_set(self.session_pool_def, "properties", "customContainerTemplate", value=customer_container_template)
+        safe_set(self.session_pool_def, "properties", "secrets", value=secrets_def)
+        safe_set(self.session_pool_def, "properties", "dynamicPoolConfiguration", value=dynamic_pool_def)
+        safe_set(self.session_pool_def, "properties", "sessionNetworkConfiguration", value=session_network_def)
+        safe_set(self.session_pool_def, "properties", "scaleConfiguration", value=session_scale_def)
```

Comment:
You didn't move this part in to set_up_xxx() if it can as we sync offline.
I think it would better to move into set_up_xxx() to be consistent with others.

Reply:
Yes, for create part, I remain this. I will raise another Pr to change this. Let's make this PR in first.